staging/src/k8s.io/apiserver/pkg/endpoints/handlers/namer.go

```diff
@@ -100,15 +100,6 @@ func (n ContextBasedNaming) GenerateLink(req *http.Request, obj runtime.Object)
 		return "", fmt.Errorf("missing requestInfo")
 	}
 
-	if len(namespace) == 0 && len(name) == 0 {
-		if len(requestInfo.Name) == 0 {
```

Comment:
That's not quite the same as `ObjectName`, since `ObjectName` verifies just the `name` here but no `namespace`. Additionally there's a `requestInfo.Name` check here, which will be removed with your change. I don't think this is proper change. @deads2k might confirm, if I'm wrong.